.github/workflows/runtime-upgrade.yml

```diff
@@ -0,0 +1,809 @@
+
+      - name: TERM creditcoin-node
+        continue-on-error: true
+        run: |
+          killall -TERM creditcoin-node
+          sleep 120
+
+      - name: KILL creditcoin-node
+        continue-on-error: true
+        run: |
+          killall -KILL creditcoin-node
+          sleep 120
+
+      - name: Upload logs
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: logs
+          path: "*.log"
+
+      - name: Upload creditcoin-fork.json
+        uses: actions/upload-artifact@v3
+        with:
+          name: creditcoin-fork.json
+          path: creditcoin-fork.json
+
+  live-sync-creditcoin:
+    needs:
+      - setup
+      - deploy-github-runner
+    runs-on:
+      [self-hosted, "${{ needs.deploy-github-runner.outputs.runner_vm_name }}"]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Download creditcoin-node for release ${{ needs.setup.outputs.release_tag }}
+        uses: i3h/download-release-asset@v1
+        with:
+          owner: gluwa
+          repo: creditcoin
+          tag: ${{ needs.setup.outputs.release_tag }}
+          file: ${{ needs.setup.outputs.artifact_name }}
+
+      - name: Sync with ${{ needs.setup.outputs.target_chain }} past block number ${{ needs.setup.outputs.last_block_number }}
+        run: |
+          # see https://opensource.com/article/18/5/how-find-ip-address-linux
+          IP_ADDRESS=$(curl https://ifconfig.me)
+          echo "INFO: IP_ADDRESS=$IP_ADDRESS"
+
+          sudo apt-get update
+          sudo apt install -y unzip
+
+          unzip creditcoin-*-unknown-linux-gnu.zip
+          chmod a+x ./creditcoin-node
+
+          ./creditcoin-node --version
+          ./creditcoin-node \
+            --name "test-node-$GITHUB_RUN_ID-attempt-$GITHUB_RUN_ATTEMPT" \
+            --chain ${{ needs.setup.outputs.target_chain }} \
+            --validator --mining-key 5DkPYq8hFiCeGxFBkz6DAwnTrvKevAJfTYrzFtr9hpDsEAU1 \
+            --bootnodes "${{ needs.setup.outputs.boot_node }}" \
+            --prometheus-external --pruning archive \
+            --telemetry-url "wss://telemetry.creditcoin.network/submit/ 0" \
+            --public-addr "/dns4/$IP_ADDRESS/tcp/50555" \
+            --port 50555 >creditcoin-node-initial-live-sync.log 2>&1 &
+
+      - name: Wait for creditcoin-node to sync past block number ${{ needs.setup.outputs.last_block_number }}
+        run: |
+          ./scripts/wait-for-sync.sh ${{ needs.setup.outputs.last_block_number }}
+
+      - name: TERM creditcoin-node
+        continue-on-error: true
+        run: |
+          killall -TERM creditcoin-node
+          sleep 120
+
+      - name: KILL creditcoin-node
+        continue-on-error: true
+        run: |
+          killall -KILL creditcoin-node
+          sleep 120
+
+      - name: Upload logs
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: logs
+          path: "*.log"
+
+  test-against-fork:
```

Comment:
What about this ?

```
2023-08-24 05:21:31 Creditcoin Node    
2023-08-24 05:21:31 ✌️  version 2.230.0-31b85661f67    
2023-08-24 05:21:31 ❤️  by Gluwa Inc.:Nathan Whitaker <nathan.whitaker@gluwa.com>, 2017-2023    
2023-08-24 05:21:31 📋 Chain specification: Development    
2023-08-24 05:21:31 🏷  Node name: Alice    
2023-08-24 05:21:31 👤 Role: AUTHORITY    
2023-08-24 05:21:31 💾 Database: ParityDb at /home/ubuntu/.local/share/creditcoin-node/chains/creditcoin-fork/paritydb/full    
2023-08-24 05:21:31 ⛓  Native runtime: creditcoin-node-230 (creditcoin-node-0.tx11.au2)    
2023-08-24 05:22:14 🔨 Initializing Genesis block/state (state: 0x3ecf…3afb, header-hash: 0x932b…7613)    

====================

Version: 2.230.0-31b85661f67

   0: sp_panic_handler::set::{{closure}}
   1: call<(&core::panic::panic_info::PanicInfo), (dyn core::ops::function::Fn<(&core::panic::panic_info::PanicInfo), Output=()> + core::marker::Send + core::marker::Sync), alloc::alloc::Global>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/alloc/src/boxed.rs:1999:9
      rust_panic_with_hook
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:709:13
   2: {closure#0}
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:597:13
   3: __rust_end_short_backtrace<std::panicking::begin_panic_handler::{closure_env#0}, !>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/sys_common/backtrace.rs:151:18
   4: begin_panic_handler
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:593:5
   5: panic_fmt
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/core/src/panicking.rs:67:14
   6: panic_display<&str>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/core/src/panicking.rs:150:5
      panic_str
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/core/src/panicking.rs:134:5
      expect_failed
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/core/src/option.rs:1952:5
   7: creditcoin_node::service::new_partial
   8: creditcoin_node::service::new_full
   9: tokio::runtime::park::CachedParkThread::block_on
  10: tokio::runtime::context::runtime::enter_runtime
  11: tokio::runtime::runtime::Runtime::block_on
  12: sc_cli::runner::Runner<C>::run_node_until_exit
  13: creditcoin_node::command::run
  14: creditcoin_node::main
  15: std::sys_common::backtrace::__rust_begin_short_backtrace
  16: std::rt::lang_start::{{closure}}
  17: call_once<(), (dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/core/src/ops/function.rs:284:13
      do_call<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:500:40
      try<i32, &(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe)>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:464:19
      catch_unwind<&(dyn core::ops::function::Fn<(), Output=i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe), i32>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panic.rs:142:14
      {closure#2}
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/rt.rs:148:48
      do_call<std::rt::lang_start_internal::{closure_env#2}, isize>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:500:40
      try<isize, std::rt::lang_start_internal::{closure_env#2}>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panicking.rs:464:19
      catch_unwind<std::rt::lang_start_internal::{closure_env#2}, isize>
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/panic.rs:142:14
      lang_start_internal
             at /rustc/e6d4725c76f3b526c74454bc51afdf6daf133506/library/std/src/rt.rs:148:20
  18: main
  19: <unknown>
  20: __libc_start_main
  21: _start


Thread 'main' panicked at 'No initial authorities configured for GRANDPA', node/src/service.rs:162

This is a bug. Please report it at:

	support.anonymous.an
```

Reply:
It's probably a bug in either [creditcoin-fork](https://github.com/gluwa/creditcoin-fork) or our workflow. I'm not particularly concerned about it because the job effectively just forks mainnet - but staging is already a fork of mainnet that we performed the switch on.

I can try to run this locally to make sure, it's easy to work around locally because you can just edit the forked chainspec to add the `grandpaInitialAuthorities` field. To fix it in CI we'd have to identify the bug and fix it, which will take longer.

Reply:
I talked to Nathan about this, and while it happening in the pipeline is odd, we have believe it won't affect the Mainnet release since the staging update succeeded without an issue.

Reply:
FTR: I ran this manually and once working around the bug, everything worked correctly (runtime upgrade, migrations, switch to pos). 

I didn't run the integration test suite against the fork as that didn't seem valuable given the fact that we've already run them against the new runtime.